cpp/turbodbc/Library/src/parameter_sets/set_field.cpp

```diff
@@ -65,36 +66,41 @@ namespace {
 
 		void operator()(double const & value)
 		{
-			*reinterpret_cast<double *>(destination_.data_pointer) = boost::get<double>(value);
+			*reinterpret_cast<double *>(destination_.data_pointer) = value;
 			destination_.indicator = sizeof(double);
 		}
 
-		void operator()(boost::posix_time::ptime const & value)
+		void operator()(std::chrono::system_clock::time_point const & value)
 		{
-			auto const & date = value.date();
-			auto const & time = value.time_of_day();
-			auto destination = reinterpret_cast<SQL_TIMESTAMP_STRUCT *>(destination_.data_pointer);
-
-			destination->year = date.year();
-			destination->month = date.month();
-			destination->day = date.day();
-			destination->hour = time.hours();
-			destination->minute = time.minutes();
-			destination->second = time.seconds();
-			// map posix_time microsecond precision to SQL nanosecond precision
-			destination->fraction = time.fractional_seconds() * 1000;
+			auto tp = value;
+			auto dp = std::chrono::floor<std::chrono::days>(tp);
+			std::chrono::year_month_day ymd{dp};
+			auto time_of_day = tp - dp;
+			auto hrs   = std::chrono::duration_cast<std::chrono::hours>(time_of_day);
+			time_of_day -= hrs;
+			auto mins  = std::chrono::duration_cast<std::chrono::minutes>(time_of_day);
+			time_of_day -= mins;
+			auto secs  = std::chrono::duration_cast<std::chrono::seconds>(time_of_day);
+			time_of_day -= secs;
+			auto micros = std::chrono::duration_cast<std::chrono::microseconds>(time_of_day);
 
+			auto destination = reinterpret_cast<SQL_TIMESTAMP_STRUCT *>(destination_.data_pointer);
+			destination->year = static_cast<SQLSMALLINT>(int(ymd.year()));
```

Comment:
No intention, or time to review the entire PR. However I can point out that:

`int ( ... )` is an old C style syntax for a static cast.
`static_cast<...>(...)` is the C++ native syntax for static cast.

So what is written here is a static cast, nested inside a static cast. If it is required likely only the outer one is needed.

Cheers, Markus

Reply:
Thanks for having a look at it!
Looks like the nested casts were needed, with the outer cast only it didn't compile. Made the syntax consistent.